datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/JSR310DeserializerBase.java

```diff
@@ -141,6 +141,13 @@ protected boolean _isValidTimestampString(String str) {
     @SuppressWarnings("unchecked")
     protected <R> R _handleDateTimeException(DeserializationContext context,
               DateTimeException e0, String value) throws JsonMappingException
+    {
+        return _handleWeirdStringValue(context, e0, value);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected <R, E extends RuntimeException> R _handleWeirdStringValue(DeserializationContext context,
```

Comment:
No need to add type parameter `E`, I think, ok to just pass `Exception`, I think.